### PR TITLE
Add new `Layout` design page

### LINKED ISSUE #8561: [Layout training] Create Layout resource page
Create a Layout training resource page under the `Design` category on polaris.shopify.com

[Google doc](https://docs.google.com/document/d/1MdHxd-KvmrIxIYKw8ZM_f6sGdKD8j-9r314CHajDAAk/edit#) for content writing

In sequence order:

```[tasklist]
### Tasks
- [ ] https://github.com/Shopify/polaris/issues/8633
- [ ] https://github.com/Shopify/polaris/issues/8562
- [ ] https://github.com/Shopify/polaris/issues/8637
- [ ] https://github.com/Shopify/polaris/issues/8632
- [ ] https://github.com/Shopify/polaris/issues/8563
- [ ] https://github.com/Shopify/polaris/issues/8564
- [ ] https://github.com/Shopify/polaris/issues/8635
- [ ] https://github.com/Shopify/polaris/issues/8634
```

### PR DESCRIPTION
**Pull request by yurm04**
### WHY are these changes introduced?

Fixes #8561  

### WHAT is this pull request doing?
Adding a new `Layout` resource page to the `Design` section in the polaris.shopify.com site
localhost_3000_design_layout (1)

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris.shopify.com/content/design/layout.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+description: Polaris layout components let you quickly build custom layouts without writing CSS code.
+---
+
+---
+
+<!-- <div>
+  <img src="/images/foundations/design/layout/layout-banner.png" alt="A visual example of a polaris unit">
+</div> -->
+
+![Building blocks in various combinations coming together to make up the same overall structure](/images/foundations/design/layout/layout-banner.png)
```
Comment (yurm04) [inline polaris.shopify.com/content/design/layout.md]: @jjgali mind giving all of the alt text an edit pass? 🙏

File: polaris.shopify.com/content/design/layout.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+<video width="100%" height="auto" controls autoplay muted loop>
+  <source src="/images/foundations/design/space/breakpoints.mp4" type="video/mp4">
+  <p>Browser window resizing with the overlayed column grid dynamically adjusting to the size</p>
+</video>
+
+To ensure designs are responsive, reference the breakpoints and examples in the Polaris [Space](/design/space) documentation.
+
+## Layout component examples
+
+The following examples illustrate how the new layout components can be composed together to create admin UI elements. Each example is just one approach to building that particular bit of UI, but many other combinations of layout components can result in the same outcome. To get a feel for these new components, try to rebuild the same UI using a different combination of our new layout components.
```
Comment (yurm04) [inline polaris.shopify.com/content/design/layout.md]: @jjgali added this blurb here to try and make clearer what these examples were for. Please feel free to edit it, I kind of wrote it on the fly last minute

File: polaris.shopify.com/content/design/layout.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+keywords:
+  - layout
+  - layout components
+  - structure
+  - position
+  - stack
+  - columns
+  - bleed
+  - card
+  - Box
```
Comment (laurkim) [inline polaris.shopify.com/content/design/layout.md]:
```suggestion
  - box
  - inline
  - divider
  - responsive
  - spacing
```

**Review by jjgali — APPROVED** [review]
LGTM! Content review finished.

**Review by laurkim — APPROVED** [review]
Nice!! 💯 🥇 :shipit:

**Review by tracytees — APPROVED** [review]

**Pull request merged**